internal/cmd/skupper/connector/nonkube/connector_update.go

```diff
@@ -2,28 +2,168 @@ package nonkube
+	"github.com/skupperproject/skupper/internal/nonkube/client/fs"
+	"github.com/skupperproject/skupper/pkg/apis/skupper/v1alpha1"
+	"github.com/skupperproject/skupper/pkg/utils/validator"
 	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type ConnectorUpdates struct {
+	routingKey    string
+	host          string
+	connectorType string
+	port          int
+	tlsSecret     string
+}
 type CmdConnectorUpdate struct {
-	CobraCmd  *cobra.Command
-	Flags     *common.CommandConnectorUpdateFlags
-	Namespace string
-	siteName  string
+	connectorHandler *fs.ConnectorHandler
+	CobraCmd         *cobra.Command
+	Flags            *common.CommandConnectorUpdateFlags
+	namespace        string
+	connectorName    string
+	newSettings      ConnectorUpdates
+	output           string
 }
 
 func NewCmdConnectorUpdate() *CmdConnectorUpdate {
 	return &CmdConnectorUpdate{}
 }
 
 func (cmd *CmdConnectorUpdate) NewClient(cobraCommand *cobra.Command, args []string) {
-	//TODO
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace) != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String() != "" {
+		cmd.namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
+	}
+
+	cmd.connectorHandler = fs.NewConnectorHandler(cmd.namespace)
+}
+
+func (cmd *CmdConnectorUpdate) ValidateInput(args []string) []error {
+	var validationErrors []error
+	resourceStringValidator := validator.NewResourceStringValidator()
+	numberValidator := validator.NewNumberValidator()
+	connectorTypeValidator := validator.NewOptionValidator(common.ConnectorTypes)
+	outputTypeValidator := validator.NewOptionValidator(common.OutputTypes)
+
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameContext) != nil && cmd.CobraCmd.Flag(common.FlagNameContext).Value.String() != "" {
+		fmt.Println("Warning: --context flag is not supported on this platform")
+	}
+
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameKubeconfig) != nil && cmd.CobraCmd.Flag(common.FlagNameKubeconfig).Value.String() != "" {
+		fmt.Println("Warning: --kubeconfig flag is not supported on this platform")
+	}
+
+	// Validate arguments name
+	if len(args) < 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("connector name must be configured"))
+	} else if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("connector name must not be empty"))
+	} else {
+		ok, err := resourceStringValidator.Evaluate(args[0])
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("connector name is not valid: %s", err))
+		} else {
+			cmd.connectorName = args[0]
+		}
+	}
+
+	// Validate that there is already a connector with this name in the namespace
+	if cmd.connectorName != "" {
+		connector, err := cmd.connectorHandler.Get(cmd.connectorName)
```

Comment:
I believe that the resource handlers should return resources from the `input/sources` directory,
when the `runtime/state` directory does not exist for a given namespace.

The reason is because, returning from `runtime/state` only works if a namespace has already been
bootstrapped. If the user is using the CLI to prepare the site state, but haven't yet initialized it,
this will cause an error, while the resource is already placed as an input source.

Here is an example:

```
$ ./skupper connector create myservice 8080
File written to /home/user/.local/share/skupper/namespaces/default/input/sources/connectors/myservice.yaml
$ ./skupper connector status
No connectors found:
failed to read directory: open /home/user/.local/share/skupper/namespaces/default/runtime/state/connectors: no such file or directory
```

The absence of the `runtime` directory means that the given namespace was not yet bootstrapped.

Maybe if a namespace hasn't been initialized we should read from `input/sources` and warn about the
site not yet initialized? 

In case the namespace is initialized, there will be two source directories: `sources` and `input/sources`.
The first represents the current sources loaded during bootstrap, while the latter represents the desired
state, provided by the user (usually using the CLI).

If they differ, it means that user has made changes to the `input/sources` and there are pending changes to
be processed by a forced bootstrap execution.

Reply:
done

Reply:
I agree with the warning of  "Site not initialized yet" and to search in the input directory as well, but can we move the warning to the implementation of the Get command and just return a value or an error? I think that logic is part of how Get search for resources.

Reply:
done

Reply:
I think that the List() is not yet included in the Router Access Handler

Reply:
updated,  thanks missed that.